src/hooks/useTextFilter.ts

```diff
@@ -1,35 +1,28 @@
-import {useEffect, useState} from "react";
-import {Filters} from "@/routes/events";
-import {FilterController} from "@/hooks/useDateFilter";
+import {useCallback, useState} from "react";
+import {FilterController, UseFilterHook} from "@/types/filtering.types";
 
 export interface TextFilterController extends FilterController {
-    setSearchText: (searchText: string) => void;
-    searchText: string;
+  setSearchText: (searchText: string) => void;
+  searchText: string;
 }
 
-/**
- *
- * @param {string} paramName the name of the search parameter that will be used to filter the data.
- *      This value will be appended to the query url.
- * @returns DateFilterController
- */
-export const useTextFilter = (paramName: string, filters: [string,string][], setFilters: (filters: [string,string][]) => void) => {
-    const [searchText, setSearchText] = useState<string>('');
+const queryDefinition = "filters[$and][0][title][$contains]";
```

Comment:
currently title is correct, but we might want to pass the exact name of the property we filter on in the future
but this is good for now. just a thing to keep in mind